vcl/main.vcl

```diff
@@ -135,6 +135,13 @@ sub vcl_fetch {
         return (pass);
     }
 
+    # If we've gotten an error after the restarts on an ESI page, then we'll
+    # replace the error contents with a blank page so that we don't render the
+    # error contents into the final document.
+    if (req.url ~ "^/_esi/" && (beresp.status == 500 || beresp.status == 503)) {
```

Comment:
``beresp.http.Warehouse-ESI-Enable`` only exists on the primary response (it's how we signal to Varnish to enable ESI for this response). Good point on the order of the if statements though.